src/stats.rs

```diff
@@ -544,9 +544,9 @@ pub(crate) fn convert_time_intervals(
         .unwrap()
         .map(|v| {
             // The first interval of each day should start with 00:00:00.
-            let mut ts = v / ((24 * 60 * 60) * (24 * 60 * 60) * A_BILLION);
-            ts += (v - ts) / time_interval * time_interval;
-            NaiveDateTime::from_timestamp_opt(ts, 0).unwrap_or_default()
+            let mut interval_idx = v / A_BILLION;
```

Comment:
In my perspective, the following is the intention of the original code:
```Rust
{
  let A_DAY = 24 * 60 * 60;
  let start_of_day = v / A_BILLION / A_DAY * A_DAY;
  let ts = (v - start_of_day) / time_interval * time_interval;
  NaiveDateTime::from_timestamp_opt(ts, 0).unwrap_or_default()
}
```
As the comments states, it intended to keep the first interval as [00:00:00, 00:00:00+ time_interval] for each day. The modified code has removed such restriction. 
- Should it be intentional, I think the comments should be removed and add the change to the changelog for record keeping.
- Otherwise, corresponding modification is needed. 
- Further, a unit test regarding the first interval of the day would be appreciated.

Reply:
The original code have a problem.
```rust
    let start_of_day = v / A_BILLION / A_DAY * A_DAY;          # the number of day except below seconds values
    let ts = (v - start_of_day)/ time_interval * time_interval;  # the time beginning from 1970-01-01
```

As an example, for the input datetime `2019-06-22 00:09:20.123456789`
* With original code, `ts` will be `1970-01-01 00:00:00.000000000` if time_interval is 600.
* The correct `ts` value is `2019-06-22 00:00:00.000000000` to draw chart normally in Web UI.

The chart with the modified code works ok:
![perf](https://github.com/petabi/structured/assets/68586156/f91d5e68-959a-4117-a28b-261774db376e)